src/main/java/uk/gov/pay/webhooks/webhook/WebhookService.java

```diff
@@ -157,19 +158,32 @@ private boolean webhookHasSubscriptionForEvent(WebhookEntity webhook, InternalEv
                 .anyMatch(internalEventNames -> internalEventNames.contains(event.eventType()));
     }
 
-    public void deleteWebhookMessagesOlderThan() {
+    public void expireWebhookMessages() {
         int maxAgeOfMessages = webhookMessageDeletionConfig.getMaxAgeOfMessages();
         int maxNumOfMessagesToExpire = webhookMessageDeletionConfig.getMaxNumOfMessagesToExpire();
+
+        List<WebhookMessageEntity> candidateWebhookMessagesForDeletion = webhookMessageDao.getWebhookMessagesOlderThan(maxAgeOfMessages);
```

Comment:
Any reason we are querying for a list of IDs and deleting webhook messages?

Can we not delete webhook_messages and cascade by date?

Reply:
🤦 Delete cascade would be much easier. Might even warrant an entirely new PR.